packages/firebase_auth/firebase_auth/lib/src/user.dart

```diff
@@ -421,6 +421,10 @@ class User {
     await _delegate.verifyBeforeUpdateEmail(newEmail, actionCodeSettings);
   }
 
+  MultiFactor get multiFactor {
+    return MultiFactor._(_delegate.multiFactor);
```

Comment:
MultiFactor either should override == or this property should be cached.

At the moment `this.multiFactor != this.multiFactor`

Reply:
Good catch, thanks